earthpy/io.py

```diff
@@ -117,51 +136,55 @@ def get_data(self, key=None, replace=False, url=None):
         -------
         path_data : str
             The path to the downloaded data.
+
+        Examples
+        --------
+        Download a dataset using a key:
+
+            >>> earthpy.data.get_data('california-rim-fire') # doctest: +SKIP
+
+        Or, download a dataset using a figshare URL:
+
+            >>> url = 'https://ndownloader.figshare.com/files/12395030'
+            >>> earthpy.data.get_data(url=url) # doctest: +SKIP
+
         """
         if key is not None and url is not None:
             raise ValueError(
                 "The `url` and `key` parameters can not both be "
                 "set at the same time."
             )
         if key is None and url is None:
-            print("Available datasets: {}".format(list(DATA_URLS.keys())))
+            print(self.__repr__())
             return
 
         if key is not None:
             if key not in DATA_URLS:
-                raise ValueError(
-                    "Don't understand key "
+                raise KeyError(
+                    "Key " + key + " not found in earthpy.io.DATA_URLS. "
                     "{}\nChoose one of {}".format(key, DATA_URLS.keys())
                 )
 
             this_data = DATA_URLS[key]
             this_root = op.join(self.path, key)
 
         if url is not None:
-            # try and workout the filename and file type
-            fname = None
-            r = requests.head(url)
-            content_disposition = r.headers["content-disposition"].split(";")
-            for c in content_disposition:
-                if c.startswith("filename="):
-                    fname = c.split("=")[1]
-                    break
-            else:
-                raise RuntimeError(
-                    "Could not deduce filename for " "{}.".format(url)
-                )
+            with requests.get(url) as r:
+                if "content-disposition" in r.headers.keys():
+                    content = r.headers["content-disposition"]
+                    fname = re.findall("filename=(.+)", content)[0]
+                else:
+                    fname = url.split("/")[-1]
 
             # try and deduce filetype
             file_type = "file"
             for kind in ALLOWED_FILE_TYPES:
                 if fname.endswith(kind):
                     file_type = kind
 
-            # strip off the file extension so we get pretty download
-            # directories
+            # strip off file extension so we get pretty download directories
             if file_type != "file":
-                # cut off an extra character to remove the trailing dot as well
-                fname = fname[: -(len(file_type) + 1)]
+                fname = os.path.splitext(fname)[0]
 
             this_data = (url, fname, file_type)
             this_root = op.join(self.path, "unsorted")
```

Comment:
i'm playing with this now but i'm not sure why an `unsorted` directory is created. Perhaps we can ask for a url and a dir_name to allow the user to specify a dirname of their choice?

Reply:
i tried this
1. add dir_name as an optional var in the class
then
```
if not dir_name:
    dir_name = "data-download"
this_root = op.join(self.path, dir_name)
```

Reply:
Based on our Slack convo we will change the default download directory for unsorted downloads to "earthpy-downloads/"

Reply:
Changed in 6f279daaf765204e9249521c96b368511d80d718